packages/wp-source/src/libraries/handlers/postType.ts

```diff
@@ -33,11 +74,24 @@ const postTypeHandler = ({
 
       // exit loop if this endpoint returns an entity!
       if (populated.length > 0) {
-        isHandled = true;
-        break;
+        // We have to check if the link property in the data that we
+        // populated is the same as the current route.
+        if (populated[0].link === route) {
+          isHandled = true;
+          break;
+        } else {
+          isMismatched = true;
```

Comment:
You're right. But now I've realized that probably we should do:

```js
if (populated[0].link === route) {
  isHandled = true;
  isMismatched = false; // <--- added this
  break;
} else {
  isMismatched = true;
}
```

Because it might happen that 
- we match a route,
- `api.get()` the entity e.g. `post`
- that entity was matched "by accident" `(populated[0].link !== route)`
- we `api.get()` another entity, e.g. `media`
- this entity is the "correct" one (the link matches)
- so we should not throw an error.

Reply:
Right, right. Yeah, let's implement it that way.